### PR TITLE
Create implementation_plan

### PR DESCRIPTION
**Pull request by james-mm**
This will be used for new customers as an easy-to-use guide for implementing Mattermost

Another pull request will follow to include this file under the Mattermost Overview page (https://docs.mattermost.com/overview/index.html).

This is a first iteration of this file, more information will be added on over time.

### CODE REVIEW

**Comment by jasonblais** [comment]
This is great.

@james-mm Is the goal of the document so that once the customer has run through it, they'll have all the proper systems in place (e.g. performance monitoring)? What else is planned in future?

@lindy65 Would you be able to help convert this document to .rst first?

**Comment by jasonblais** [comment]
@amyblais Actually, would you be able to help?

**Comment by amyblais** [comment]
I converted it to .rst and I'll fix formatting during today.

**Comment by james-mm** [comment]
@amyblais Thanks for making the updates!

@jasonblais Yes, the customer should have most of the basic functions in place. It is about 90% complete, some sections direct users to links to set up enterprise features (AD/LDAP, HA, SAML) but not all (performance, compliance export, elasticsearch).

**Comment by amyblais** [comment]
@jasonblais Formatting is done ^

**Comment by james-mm** [comment]
@amyblais Thank you! Making a mental note to add future docs in rst format 👍

File: source/overview/implementation_plan.rst [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+   - Mattermost users can use their favorite Web browser to connect to their Mattermost server in place of the Mattermost mobile and desktop applications.
+ - **Proxy Server**
+   - The use of a proxy server (NGINX, Apache, HAProxy, ELB, etc.) is recommended with Mattermost to obtain the benefits of: SSL termination, HTTP to HTTPS redirection, port mapping :80 to :8065, and the creation of standard request logs.
+
+The following diagram is a high-level illustration of the Mattermost platform and how its various components interact with each other within a network:
+
+.. image:: ../images/network.PNG
+**Reference**: https://docs.mattermost.com/deployment/deployment.html
+
+A minimal installation of Mattermost Enterprise E20 for up to 500 users with application (Mattermost Server) and database high availability features enabled will require the following physical, or virtual, servers:
```
[the remark this answers is not in the document]
Comment (jasonblais) [reply source/overview/implementation_plan.rst]: Also, what would be the recommendations for a larger install?

File: source/overview/implementation_plan.rst [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+
+ - Team Edition
+
+ - Enterprise Edition (E10) which adds additional features to the platform including AD/LDAP integration, multi-factor authentication ("MFA"), custom branding, advanced access control policy and next business day support.
+
+ - Enterprise Edition (E20) includes E10 features plus security controls, compliance reporting, high availability, and enterprise class support. 
+
+The purpose of this Implementation Plan is to:
+
+ - Describe the solution to be deployed;
```
Comment (jasonblais) [inline source/overview/implementation_plan.rst]: At first it was hard to make an association that `Describe the solution to be deployed` was section 1.2, `Outline the process for implementing the solution (key personnel, tasks, schedule, etc.) was section 2, etc.

One option might to re-number 1.2 --> 2, 2 --> 3, 3 --> 4 to make it more clear, or consider matching these bullet points with the section headings to reinforce the connection.

I'm 0/5 though.

File: source/overview/implementation_plan.rst [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+
+Mattermost is an open source, private cloud alternative to proprietary SaaS messaging solutions designed to offer organizations superior control, data sovereignty, configurability, freedom from lock-in, and enhanced security by keeping vital communications behind your firewall. You have chosen to implement Mattermost in our supported `editions:`_
+
+.. _`editions:`: https://about.mattermost.com/features/
+
+ - Team Edition
+
+ - Enterprise Edition (E10) which adds additional features to the platform including AD/LDAP integration, multi-factor authentication ("MFA"), custom branding, advanced access control policy and next business day support.
+
+ - Enterprise Edition (E20) includes E10 features plus security controls, compliance reporting, high availability, and enterprise class support.
```
[the remark this answers is not in the document]
Comment (james-mm) [reply source/overview/implementation_plan.rst]: Updated

File: source/overview/implementation_plan.rst (hunk 2 above)
Comment (james-mm) [reply source/overview/implementation_plan.rst]: Updated

File: source/overview/implementation_plan.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (james-mm) [reply source/overview/implementation_plan.rst]: The recommendation in this doc is for HA whereas the one in the link is on a single server.

We link out to the larger rec on the doc, but I'm open: https://docs.mattermost.com/install/requirements.html#hardware-requirements. I'll defer to you and @cvitter on what the recommended specs would be for a larger install in this doc.

File: source/overview/implementation_plan.rst [hunk 4]
```diff
@@ -0,0 +1,369 @@
...
+
+3.1      Hardware, Software, Facilities, and Materials
+.........................................
+
+In this section, list support software, materials, equipment, and facilities required for the implementation, if any.
+
+3.1.1     Hardware
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+This section provides a list of support equipment and includes all hardware used for testing time implementation.  For example, if a client/server database is implemented on a LAN, a network monitor or “sniffer” might be used, along with test programs. to determine the performance of the database and LAN at high-utilization rates.  If the equipment is site-specific, list it in Section 4, Implementation Requirements by Site.
```
[the remark this answers is not in the document]
Comment (james-mm) [reply source/overview/implementation_plan.rst]: I think this was meant as freetext for the customer to fill in additional information

File: source/overview/implementation_plan.rst [hunk 5]
```diff
@@ -0,0 +1,369 @@
...
+Present a training curriculum listing the courses that will be provided, a course sequence. and a proposed schedule.  If appropriate, identify which courses particular types of staff should attend by job position description.
+ 
+If training will be provided by one or more commercial vendors, identify them, the course name(s), and a brief description of the course content.
+ 
+If the training will be provided by State staff, provide the course name(s) and an outline of the content of each course.  Identify the resources, support materials, and proposed instructors required to teach the course(s).
+
+3.3      Performance Monitoring (E20)
+.........................................
+
+This section describes the performance monitoring tool and techniques and how it will be used to help decide if the implementation is successful.
```
[the remark this answers is not in the document]
Comment (james-mm) [reply source/overview/implementation_plan.rst]: Updated

File: source/overview/implementation_plan.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (cvitter) [reply source/overview/implementation_plan.rst]: Sorry for the delay in responding. Just seeing this notification.

Having reviewed the docs again I wouldn't have an issue with bumping the system specs we recommend for 500 users. The idea was to provide an initial base install guidance. Not having a lot of real world production install experience I don't know the right numbers here.

**Comment by jasonblais** [comment]
Thanks all! I've queued this documented for Thursday's PM meeting for a quick review. I'll make further changes/comments if needed and this is good to merge otherwise.

File: source/overview/implementation_plan.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (jasonblais) [reply source/overview/implementation_plan.rst]: @james-mm @cvitter I updated the pre-text of the table to

```
A recommended installation of Mattermost Enterprise E20 configured with a redundant, highly available, highly scalable mode, with application (Mattermost Server) and database high availability features enabled, will require the following physical, or virtual, servers:
```

In other words, I removed the recommended user count and we can use this as a good initial base install guide. They can refer to https://docs.mattermost.com/install/requirements.html#hardware-requirements for more detailed requirements.

Let me know if there are any concerns with that change.

File: source/overview/implementation_plan.rst [hunk 6]
```diff
@@ -0,0 +1,369 @@
...
+
+2.2  System Organization
+.........................................
+
+A Mattermost Enterprise E20 implementation consists of the following system components:
+
+ - **Mattermost Server**
+   - The Mattermost Server is a single binary that includes the RESTful JSON web service, authentication client, authentication provider, notification service, and data management service. The Mattermost Server can be deployed in stand-alone or high availability mode where two or more servers are clustered together using gossip protocol and a proxy server that routes traffic from client applications to healthy servers in the cluster.
+ - **Mattermost Database**
+   - A MySQL or PostgreSQL database in stand-alone or high-availability configurations (master with read replicas). When choosing the database, consider the `inherent search limitations of the database type <https://docs.mattermost.com/install/requirements.html#database-software>`_ and whether `Elasticsearch <https://docs.mattermost.com/deployment/elasticsearch.html>`_ is most appropriate for your deployment.
```
Comment (jasonblais) [inline source/overview/implementation_plan.rst]: @james-mm Added a brief note about the inherent search limitations of the database type, and that Elasticsearch may sometimes be the most appropriate option for E20 customers.

**Comment by james-mm** [comment]
@jasonblais Awesome, can't wait for it to get pushed!

**Comment by james-mm** [comment]
@jasonblais Are there any updates on the status of this PR?

**Comment by jasonblais** [comment]
@james-mm Whoops, sorry. I meant to merge it. Thanks for the ping, merging now 👍

**Comment by james-mm** [comment]
@jasonblais Thanks!

**Pull request merged**